src/meta/src/manager/catalog/fragment.rs

```diff
@@ -267,48 +267,69 @@ where
 
     /// Drop table fragments info and remove downstream actor infos in fragments from its dependent
     /// tables.
-    pub async fn drop_table_fragments(&self, table_id: &TableId) -> MetaResult<()> {
+    pub async fn drop_table_fragments_vec(&self, table_ids: &HashSet<TableId>) -> MetaResult<()> {
         let map = &mut self.core.write().await.table_fragments;
+        let table_fragments_vec = table_ids
+            .iter()
+            .map(|table_id| map.get(table_id).unwrap())
+            .collect_vec();
 
-        if let Some(table_fragments) = map.get(table_id) {
-            let mut transaction = Transaction::default();
+        let mut transaction = Transaction::default();
+        for table_fragments in &table_fragments_vec {
             table_fragments.delete_in_transaction(&mut transaction)?;
+        }
 
-            let dependent_table_ids = table_fragments.dependent_table_ids();
-            let chain_actor_ids = table_fragments.chain_actor_ids();
-            let mut dependent_tables = Vec::with_capacity(dependent_table_ids.len());
-            for dependent_table_id in dependent_table_ids {
-                let mut dependent_table = map
-                    .get(&dependent_table_id)
-                    .ok_or_else(|| anyhow!("table_fragment not exist: id={}", dependent_table_id))?
-                    .clone();
-                for fragment in dependent_table.fragments.values_mut() {
-                    if fragment.fragment_type == FragmentType::Sink as i32 {
-                        for actor in &mut fragment.actors {
-                            // Remove these downstream actor ids from all dispatchers.
-                            for dispatcher in &mut actor.dispatcher {
-                                dispatcher
-                                    .downstream_actor_id
-                                    .retain(|x| !chain_actor_ids.contains(x));
-                            }
-                            // Remove empty dispatchers.
-                            actor
-                                .dispatcher
-                                .retain(|d| !d.downstream_actor_id.is_empty());
-                        }
-                    }
-                }
-                dependent_table.upsert_in_transaction(&mut transaction)?;
-                dependent_tables.push(dependent_table);
-            }
+        let chain_actor_ids = table_fragments_vec
+            .iter()
+            .flat_map(|table_fragments| table_fragments.chain_actor_ids())
+            .collect_vec();
+
+        let dependent_table_ids = table_fragments_vec
+            .iter()
+            .flat_map(|table_fragments| table_fragments.dependent_table_ids())
+            .filter(|table_id| !table_ids.contains(table_id))
+            .collect::<HashSet<_>>();
+
+        let mut dependent_tables = dependent_table_ids
```

Comment:
In scenario of `drop database`, the dependent tables should be included in the drop list as well and don't have to be updated. And they will be inserted back in L317, which is not correct.
Maybe we can add a much more simple function for that, since I don't think we gonna have requirement for dropping multi streaming jobs except `drop database`. In the new function, we can directly remove all the `TableFragments` and notify fragment mapping only.

Reply:
I have filter the table_ids which is in the drop list in L290, so I think its behavior is correct.  
I think the new function you talk about is fine.

Reply:
Good, then to keep it or add a new function are both okay for me.